test/TestProjects/Models-Cadl/Generated/BaseModelWithDiscriminator.cs

```diff
@@ -5,25 +5,23 @@
 
 #nullable disable
 
-using System;
-using Azure.Core;
-
 namespace ModelsInCadl
 {
-    /// <summary> Base model with discriminator property. </summary>
-    public partial class BaseModelWithDiscriminator : BaseModel
+    /// <summary>
+    /// Base model with discriminator property.
+    /// Please note <see cref="BaseModelWithDiscriminator"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+    /// The available derived classes include <see cref="DerivedModelWithDiscriminatorA"/> and <see cref="DerivedModelWithDiscriminatorB"/>.
+    /// </summary>
+    public abstract partial class BaseModelWithDiscriminator
     {
         /// <summary> Initializes a new instance of BaseModelWithDiscriminator. </summary>
-        /// <param name="discriminatorProperty"></param>
-        /// <exception cref="ArgumentNullException"> <paramref name="discriminatorProperty"/> is null. </exception>
-        public BaseModelWithDiscriminator(string discriminatorProperty)
+        /// <param name="discriminatorProperty"> Discriminator. </param>
+        protected BaseModelWithDiscriminator(string discriminatorProperty)
```

Comment:
In HLC, base model will have an public constructor which has no discriminator as parameter. And in Mgmt, the constructor will be protected, and has no discriminator property as parameter also.

So will we change the base model constructor?

Reply:
The class is abstract so a public ctor doesn't make sense.  The only time we will keep a public ctor is IF the class was already GAd as non abstract we won't change it since that is a breaking change.  All new poly sets should work this way.  There is a way to override the existing poly sets so that autorest doesn't convert them, but it was easier to just leave hlc as non abstract for now until I get the "last code analyzer" finished.